src/sentry/receivers/onboarding.py

```diff
@@ -388,6 +362,32 @@ def record_user_context_received(project, event, **kwargs):
 
 
 event_processed.connect(record_user_context_received, weak=False)
+
+
+@first_event_with_minified_stack_trace_received.connect(weak=False)
+def record_event_with_first_minified_stack_trace_for_project(project, event, **kwargs):
+    try:
+        user: APIUser = Organization.objects.get(id=project.organization_id).get_default_owner()
+    except IndexError:
+        logging.getLogger("sentry").warning(
+            "Cannot record first event for organization (%s) due to missing owners",
+            project.organization_id,
+        )
+        return
+
+    if not project.flags.has_minified_stack_trace:
+        analytics.record(
+            "first_event_with_minified_stack_trace_for_project.sent",
+            user_id=user.id,
+            organization_id=project.organization_id,
+            project_id=project.id,
+            platform=event.platform,
+            url=dict(event.tags).get("url", None),
+        )
+
+    project.update(flags=F("flags").bitor(Project.flags.has_minified_stack_trace))
```

Comment:
I just want to note here that if a project is sending many minified stack trace events in a short period of time, then it's likely that this analytics event will fire multiple times due to a race condition

Here's how you could avoid this:

```
    # First, only enter this logic if we've never seen a minified stack trace before
    if not project.flags.has_minified_stack_trace:
        # Next, attempt to update the flag, but ONLY if the flag is currently not set. 
        # The number of affected rows tells us whether we succeeded or not. If we didn't, then skip sending the event.
        # This guarantees us that this analytics event will only be ever sent once.
        affected = Project.objects.filter(id=project.id).exclude( 
            flags=F("flags").bitand(Project.flags.has_minified_stack_trace)
        ).update(
            flags=F("flags").bitor(Project.flags.has_minified_stack_trace)
        )
        if affected:
                analytics.record(
                    "first_event_with_minified_stack_trace_for_project.sent",
                    user_id=user.id,
                    organization_id=project.organization_id,
                    project_id=project.id,
                    platform=event.platform,
                    url=dict(event.tags).get("url", None),
                )
```

(Please test this logic, I'm fairly sure it's right but make sure it updates the row as you expect)

Reply:
I've updated the code and now it should ensure that the analytic event is being sent only once.

Reply:
I have also created a test for it 😉